grype/version/gemfile_constraint_test.go

```diff
@@ -0,0 +1,40 @@
+package version
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGemfileVersionSemantic(t *testing.T) {
+	tests := []testCase{
```

Comment:
Hey @wagoodman what do you think of these test cases?

Reply:
I think so 🤔